cpp/src/detail/utility_wrappers.cu

```diff
@@ -227,5 +229,36 @@ template bool is_equal(raft::handle_t const& handle,
                        raft::device_span<int64_t const> span1,
                        raft::device_span<int64_t const> span2);
 
+template <typename data_t, bool multi_gpu>
+size_t count_values(raft::handle_t const& handle,
+                    raft::device_span<data_t const> span,
+                    data_t value)
+{
+  size_t local_count = thrust::count_if(
+    handle.get_thrust_policy(), span.begin(), span.end(), [value] __device__(data_t d) {
+      return d == value;
+    });
+
+  if constexpr (multi_gpu) {
+    local_count = cugraph::host_scalar_allreduce(
+      handle.get_comms(), local_count, raft::comms::op_t::SUM, handle.get_stream());
+  }
```

Comment:
All the other functions in this file are SG functions, and this function is an MG function. This might be confusing...

Should we better resolve this?

1. No.
2. Create another file for MG utility wrapper functions
3. Just call thrust::count() and host_scalar_allreduce() instead of creating this function.
4. Rename functions to better hint that this function performs allerduce if MG
5. Something else?

What do you think?

Reply:
I'll look at option 2.

While I like option 3, there are many cases where we don't need to create a `.cu` file except for common utilities like this.  This allows us to leave many files as `.cpp` files instead.

Reply:
Actually, I think what I'll do is move the all reduce (which doesn't use thrust, so can be called in a `.cpp` file) out of the function, making it an SG function.  So I'll leave it here but remove the `multi_gpu` flag.